### PR TITLE
gfn-electron: init at 2.1.2

### LINKED ISSUE #348122: Package request: gfn-electron
**Project description**

GeForce NOW client using electron

**Metadata**

* homepage URL: https://github.com/hmlendea/gfn-electron
* source URL: https://github.com/hmlendea/gfn-electron
* license: gpl3
* platforms: unix

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by pluiedev**
Fixes #348122

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by KiaraGrouwstra** [comment]
Result of `nixpkgs-review pr 353887` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gfn-electron</li>
  </ul>
</details>

**Comment by KiaraGrouwstra** [comment]
i can build this, but i'm a bit confused as to how to run this. (fwiw, i'm new to electron app derivations.)
the `.desktop` file mentions a `geforcenow-electron` binary, but i'm not sure i'm seeing something like that in the package.

**Comment by pluiedev** [comment]
> i can build this, but i'm a bit confused as to how to run this. (fwiw, i'm new to electron app derivations.) the `.desktop` file mentions a `geforcenow-electron` binary, but i'm not sure i'm seeing something like that in the package.

Oops I think I missed the part where I had to copy over the actual built binaries - somehow I never pushed it??

**Comment by KiaraGrouwstra** [comment]
ci complaining ^^

**Comment by KiaraGrouwstra** [comment]
that's better - the bin works for me!

(helix is nice :), mostly batteries-included but tweaking [helps too](https://codeberg.org/kiara/cfg/src/branch/main/home-manager/kiara/features/text/helix.nix). tbf it may not be as mature as neovim's ecosystem tho.)

**Comment by Scrumplex** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 353887`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>gfn-electron</li>
  </ul>
</details>

File: pkgs/by-name/gf/gfn-electron/package.nix [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    runHook postInstall
+  '';
+
+  postFixup = ''
+    makeWrapper $out/dist/geforcenow-electron $out/bin/geforcenow-electron \
+      --add-flags "--no-sandbox --disable-gpu-sandbox" \
+      --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations --enable-wayland-ime}}"
+
+    substituteInPlace $out/share/applications/com.github.hmlendea.geforcenow-electron.desktop \
+      --replace-fail "/opt/geforcenow-electron/geforcenow-electron" "$out/bin/geforcenow-electron" \
```
Comment (Scrumplex) [inline pkgs/by-name/gf/gfn-electron/package.nix:64]: See #308324
```suggestion
      --replace-fail "/opt/geforcenow-electron/geforcenow-electron" "geforcenow-electron" \
```

**Review by Scrumplex — COMMENTED** [review]
Many electron apps in Nixpkgs don't ship their own version of Electron. Perhaps we can do the same here (i.e. the wrapper should run `${lib.getExe electron}` with the .asar as the argument

**Comment by pluiedev** [comment]
> Many electron apps in Nixpkgs don't ship their own version of Electron. Perhaps we can do the same here (i.e. the wrapper should run `${lib.getExe electron}` with the .asar as the argument

Upstream explicitly doesn't build an ASAR for... *whatever reason*. It might work still, but it's not the configuration upstream supports

**Comment by Scrumplex** [comment]
> EDIT: Text appears completely blank while using an ASAR. Maybe that's why?

Could add a comment about this behaviour to the derivation? It would probably be helpful to people looking at this in the future.

**Review by Scrumplex — APPROVED** [review]

**Comment by Scrumplex** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 353887`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>gfn-electron</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>gfn-electron</li>
  </ul>
</details>

```
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
  • electron-builder  version=24.13.3 os=6.11.5-zen1
  • loaded configuration  file=package.json ("build" field)
  • writing effective config  file=dist/builder-effective-config.yaml
  • rebuilding native dependencies  dependencies=register-scheme@0.0.2 platform=linux arch=arm64
  • packaging       platform=linux arch=arm64 electron=33.0.2 appOutDir=dist/linux-arm64-unpacked
  • copying Electron  source=/nix/store/lb8qbzjfpkh0j4ngl4kgmbq2sagq92sj-electron-unwrapped-33.0.2/libexec/electron destination=/build/source/dist/linux-arm64-unpacked
  • asar usage is disabled — this is strongly not recommended  solution=enable asar and use asarUnpack to unpack files that must be externally available
  • asar usage is disabled — this is strongly not recommended  solution=enable asar and use asarUnpack to unpack files that must be externally available
buildPhase completed in 35 seconds
Running phase: installPhase
@nix { "action": "setPhase", "phase": "installPhase" }
cp: cannot stat 'dist/linux-unpacked': No such file or directory
```

---
### `aarch64-darwin`

**Comment by pluiedev** [comment]
Why the heck is it linux-arm64-unpacked... hnnnnnrgh

**Review by Scrumplex — APPROVED** [review]

**Pull request merged**